__tests__/app.test.js

```diff
@@ -8,16 +8,62 @@ beforeEach(() => seed(data));
 
 afterAll(() => db.end());
 
-describe("ANY /not-a-route", () => {
-  test("404: Responds with 404 and 'Not a valid route' when an invalid route is requested", () => {
-    return request(app)
-        .get("/api/helloworld")
-        .expect(404)
+describe("/api", () => {
+  describe("GET", () => {
+    test("Returns an object with endpoints key", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.endpoints).toBeTruthy();
+        });
+    });
+
+    test("Returns an object of objects with keys representing endpoints", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body }) => {
+          const endpoints = body.endpoints;
+          const routes = Object.keys(endpoints);
+          expect(routes.length).toBeGreaterThan(0);
+
+          validRoute = /^(GET|PUT|POST|PATCH|DELETE){1} (\/[A-Za-z:=?]*)*$/;
+          routes.forEach((route) => {
+            expect(validRoute.test(route)).toBe(true);
+          });
+        });
+    });
+
+    test("Each endpoint key is an object containing up to 4 keys description, queries, format, exampleResponse", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
         .then(({ body }) => {
-          expect(body.message).toBe("Not a valid route");
+          const endpoints = body.endpoints;
+          const routes = Object.keys(endpoints);
+          expect(routes.length).toBeGreaterThan(0);
+
+          const allowedKeys = [
+            "description",
+            "queries",
+            "format",
+            "exampleResponse",
+          ];
+          const endpointDetails = Object.values(endpoints);
+          endpointDetails.forEach((endpoint) => {
+            const endpointKeys = Object.keys(endpoint);
+
+            expect(endpointKeys.length).toBeGreaterThan(0);
+            expect(endpointKeys.length).toBeLessThanOrEqual(4);
+            endpointKeys.forEach((key) => {
+              expect(allowedKeys.includes(key)).toBe(true);
+            });
```

Comment:
We could import the endpoints.json file and check the api response against that directly :)